src/contexts/User.js

```diff
@@ -452,3 +480,45 @@ export function useUserPositions(account) {
 
   return positions
 }
+
+export function useMiningPositions(account) {
+  const [state, { updateMiningPositions }] = useUserContext()
+  const miningPositions = state?.[account]?.[MINING_POSITIONS_KEY]
+
+  const snapshots = useUserSnapshots(account)
+
+  useEffect(() => {
+    async function fetchData(account) {
+      try {
+        let miningPositionData = []
+        let result = await stakingClient.query({
+          query: MINING_POSITIONS(account),
+          fetchPolicy: 'no-cache'
+        })
+        if (!result?.data?.user?.miningPosition) {
+          return
+        }
+        miningPositionData = result.data.user.miningPosition
+        for (const miningPosition of miningPositionData) {
+          const pairAddress = miningPosition.miningPool.pair.id
+          let resultPair = await client.query({
```

Comment:
Is there a way to avoid fetching this pair data? We fetch a lot of them already so it should be populated. Maybe could use the "useAllPairs" hook to grab this stuff before, and only fetch if its not found

Reply:
done